typer/rich_utils.py

```diff
@@ -361,38 +361,47 @@ def _print_options_panel(
         opt_short_strs = []
         secondary_opt_long_strs = []
         secondary_opt_short_strs = []
+
+        # check whether argument has a metavar name or type set
+        metavar_name = None
+        metavar_type = None
+        # TODO: when deprecating Click < 8.2, make ctx required
+        signature = inspect.signature(param.make_metavar)
+        if "ctx" in signature.parameters:
+            metavar_str = param.make_metavar(ctx=ctx)
+        else:
+            # Click < 8.2
+            metavar_str = param.make_metavar()  # type: ignore[call-arg]
+        if isinstance(param, click.Argument):
+            metavar_name = metavar_str
+        if isinstance(param, click.Option):
+            metavar_type = metavar_str
+
         for opt_str in param.opts:
             if "--" in opt_str:
                 opt_long_strs.append(opt_str)
+            elif metavar_name:
+                opt_short_strs.append(metavar_name)
             else:
                 opt_short_strs.append(opt_str)
         for opt_str in param.secondary_opts:
             if "--" in opt_str:
                 secondary_opt_long_strs.append(opt_str)
+            elif metavar_name:  # pragma: no cover
+                secondary_opt_short_strs.append(metavar_name)
```

Comment:
I'm not too sure yet about these `secondary_opts`, probably requires another test (instead of having pragma cover).